packages/gasket-plugin-middleware/lib/index.d.ts

```diff
@@ -1,15 +1,73 @@
-import type { Plugin } from '@gasket/core';
-import type { FastifyReply } from "fastify";
+import type { Plugin, Gasket, MaybeAsync, MaybeMultiple } from '@gasket/core';
+import type { FastifyReply, FastifyInstance } from 'fastify';
+import type { Application as ExpressApp, Request } from 'express';
+import type {
+  Http2SecureServer,
+  Http2ServerRequest,
+  Http2ServerResponse
+} from 'http2';
+
+/** Type alias for Fastify application with HTTP/2 support */
+type FastifyApp<
+  Server = Http2SecureServer,
+  Request = Http2ServerRequest,
+  Response = Http2ServerResponse
+> = FastifyInstance<Server, Request, Response>;
+
+declare namespace Internal {
```

Comment:
Personally, I like the separation of internal types in a different file. It makes the jsdoc imports simpler and cleaner. Your change here seems to suggest the Internal namespace type will also be accessible outside the module as well, which seems not desirable. What does putting them all in the `index.d.ts` gain us?

Reply:
Ah yeah that's fair. I suppose, to me, having all of the types in one file felt easier to maintain and organize. But I'm open to keeping them separate.